bootblocks/boot_sect.S

```diff
@@ -310,19 +314,26 @@ drive_reset:
 
 //------------------------------------------------------------------------------
 
-// void disk_read (int sect, int count, byte_t * buf, int seg)
+// void disk_read_blk(int block, int sec_count, byte_t *buf, int seg)
+// void disk_read_sec(int sector, int sec_count, byte_t *buf, int seg)
+
+	.global disk_read_blk
+	.global disk_read_sec
 
-	.global disk_read
 
-disk_read:
-	// AX = sect
-	// DX = count
+disk_read_blk:	// AX = starting (1k) block 
+
+	// DX = sector count
 	// CX = buf
 	// seg pushed on stack
-	mov $ERR_DISK_READ,%bl
 
-_disk_read:
-	// BL = error code in case of failure
+	shl %ax	// turn start block into sector#, use CF later
+
+disk_read_sec:	// AX is sectors (CF must be clear!)
+
+	mov $ERR_DISK_READ,%bl // BL = error code in case of failure
+
+_disk_read:	// preset error code
```

Comment:
How many bytes do you need to reincorporate FAST_READ in to the FAT boot? I think a few more bytes can be saved by removing ERR_PAYLOAD above on 206, and calling disk_read_sec directly (this also effectively removes the need for the confusingly-named \_disk_read). I can't see any reason why a separate ERR_PAYLOAD error would ever be useful.